### PR TITLE
Dakota/federated livestreams

### PR DESCRIPTION
**Pull request by Da-Scher**
## Overall Changes
In this pull request: 
1. br3ezy now accepts requests to change federation members data. These requests are handled by sending JSON data that contains information about which br3ezy instance is asking for a change in information. This is handled with a POST request to the API link: `/api/fed/fedeIn`
*Example body of the https request to api/fed/fedeIn*
```json
{
     "fedID": 2
}
```
2. Removed sample streams for future iterations of br3ezy in anticipation of a system that creates stream database objects as the stream occurs. Requesting comment on this point.
3. br3ezy now changes database attribute `isActive` when a stream is live or not. Defaults to `isActive = FALSE`.

## Node package information
Added `circular-json` specifically to access the `stringify` method, which converts data objects into a string format that can be sent to an API.

### CODE REVIEW

**Review by joebarteluce — APPROVED** [review]
I am sad I can no longer watch Nature..

but otherwise, nice work.

**Review by justinsprecco — CHANGES_REQUESTED** [review]
Not sure how to test this, but it looks like it would work.

** See suggestions

RIP NatureStream

File: server/src/db/seeds.sql [hunk 1]
```diff
@@ -6,14 +6,13 @@ INSERT INTO Users (username, email, passwordHash, role) VALUES
...
+(1, 'Server Host Stream', 'This is the server host stream. It has the streamout.m3u8 url, so the server host can stream their zany antics to the masses.', 'https://localhost:8000/stream/streamout.m3u8', 'https://images.pexels.com/photos/169647/pexels-photo-169647.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2', FALSE);
 
 -- Insert messages
 INSERT INTO Messages (userId, streamId, body) VALUES
 (2, 1, 'Hello, world!'),
 (2, 1, 'This is a test message.');
+
+-- Insert federation
+INSERT INTO Federation (fedPublicId, apiUrl, apiPort) VALUES
+(2, '10.0.0.198', 8000, 2);
```
[the remark this answers is not in the document]
Comment (Da-Scher) [reply server/src/db/seeds.sql:18]: I see that the Federation schema should also be (fedPublicId, apiUrl, apiPort) also

File: server/src/federation/requestUpdate.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+const https = require('https');
+const mysql = require('mysql2');
+const JSON = require('circular-json');
```
[the remark this answers is not in the document]
Comment (Da-Scher) [reply server/src/federation/requestUpdate.js:3]: This is required to send the request body in a json string.
Used by line 7.

File: server/package.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
   "private": true,
   "dependencies": {
     "bcryptjs": "^2.4.3",
+    "circular-json": "^0.5.9",
```
[the remark this answers is not in the document]
Comment (Da-Scher) [reply server/package.json:14]: Going to agree actually.

File: server/src/db/seeds.sql [hunk 4]
```diff
@@ -6,14 +6,13 @@ INSERT INTO Users (username, email, passwordHash, role) VALUES
...
+(1, 'Server Host Stream', 'This is the server host stream. It has the streamout.m3u8 url, so the server host can stream their zany antics to the masses.', 'https://localhost:8000/stream/streamout.m3u8', 'https://images.pexels.com/photos/169647/pexels-photo-169647.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2', FALSE);
 
 -- Insert messages
 INSERT INTO Messages (userId, streamId, body) VALUES
 (2, 1, 'Hello, world!'),
 (2, 1, 'This is a test message.');
+
+-- Sample federation member
+--INSERT INTO Federation (fedPublicId, apiUrl, apiPort) VALUES
+--(2, '10.0.0.198', 8000, 2);
```
Comment (justinsprecco) [inline server/src/db/seeds.sql:18]:
```suggestion
-- Sample federation member
-- INSERT INTO Federation (fedPublicId, apiUrl, apiPort) VALUES
-- (2, '10.0.0.198', 8000, 2);
```

MySQL complained again, this time due to the lack of space after the comment delimiter.

**Review by justinsprecco — CHANGES_REQUESTED** [review]
Fix the seeds fix and you're approved.

**Pull request merged**